src/scaffoldmaker/meshtypes/meshtype_3d_bladder1.py

```diff
@@ -0,0 +1,770 @@
+                            result = element.setNodesByIdentifier(eft, nodeIdentifiers)
+                            elementIdentifier += 1
+
+            # create upper part of the bladder
+            for e2 in range(elementsCountUp1, (elementsCountUp1 + elementsCountUp2 - 1)):
+                for e1 in range(elementsCountAround):
+                    element = mesh.createElement(elementIdentifier, elementtemplate)
+                    bni1 = e2 * elementsCountAround + e1 + 1
+                    bni2 = e2 * elementsCountAround + (e1 + 1) % elementsCountAround + 1
+                    bni3 = bni1 + elementsCountAround
+                    bni4 = bni2 + elementsCountAround
+                    nodeIdentifiers = [bni1 + newl, bni2 + newl, bni3 + newl, bni4 + newl,
+                                       bni1, bni2, bni3, bni4]
+                    result = element.setNodesByIdentifier(eft, nodeIdentifiers)
+                    elementIdentifier += 1
+
+            # create Apex elements
+            bni3 = 1 + (elementsCountUp1 + elementsCountUp2) * elementsCountAround
+            elementtemplateApex = mesh.createElementtemplate()
+            elementtemplateApex.setElementShapeType(Element.SHAPE_TYPE_CUBE)
+            for e1 in range(elementsCountAround):
+                va = e1
+                vb = (e1 + 1) % elementsCountAround
+                eftApex = eftfactory.createEftShellPoleTop(va, vb)
+                elementtemplateApex.defineField(coordinates, -1, eftApex)
+
+                # redefine field in template for changes to eftApex:
+                element = mesh.createElement(elementIdentifier, elementtemplateApex)
+                bni1 = bni3 - elementsCountAround + e1
+                bni2 = bni3 - elementsCountAround + (e1 + 1) % elementsCountAround
+                nodeIdentifiers = [bni1 + newl, bni2 + newl, bni3 + newl, bni1, bni2, bni3]
+                result = element.setNodesByIdentifier(eftApex, nodeIdentifiers)
+                elementIdentifier += 1
+
+        fm.endChange()
+
+    @classmethod
+    def generateMesh(cls, region, options):
+        '''
+        Generate base or refined mesh.
+        :param region: Zinc region to create mesh in. Must be empty.
+        :param options: Dict containing options. See getDefaultOptions().
+        '''
+        if not options['Refine']:
+            cls.generateBaseMesh(region, options)
+            return
+
+        refineElementsCountAround = options['Refine number of elements around']
+        refineElementsCountUp = options['Refine number of elements up']
+        refineElementsCountThroughWall = options['Refine number of elements through wall']
+
+        baseRegion = region.createRegion()
+        cls.generateBaseMesh(baseRegion, options)
+
+        meshrefinement = MeshRefinement(baseRegion, region)
+        meshrefinement.refineAllElementsCubeStandard3d(refineElementsCountAround, refineElementsCountUp, refineElementsCountThroughWall)
```

Comment:
Prepare to have to return annotation groups - copy pattern from heart1 scaffold.